### PR TITLE
Fix Gen 1 Psywave mechanics

### PR DESCRIPTION
**Pull request by PlagueVonKarma**
This implements the Psywave Desync Patch from the tiering decision found [here](https://www.smogon.com/forums/threads/rby-psywave-imperfect-dvs-and-thawing-desync-glitches.3668900/)
For details on how the desync works, go [here](https://www.youtube.com/watch?v=5KmTCdnWzVI)

[Here is a replay from the RBY 2k20 PS Server showing the implementation in action.](https://replay.pokemonshowdown.com/rby-gen1customgame-897)

I'm by no means a _good_ programmer, nor do I consider myself one, so I think my code may be a bit suboptimal.

### CODE REVIEW

**Comment by PlagueVonKarma** [comment]
I realized that this would have introduced a Stadium issue, so I added a commit that prevents Desync Clause from applying there. Besides, it has a minimum of 1 damage rather than 0 there, so this should tidy it up nicely.

File: data/mods/gen1/moves.ts [hunk 1]
```diff
@@ -601,6 +601,15 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 	psywave: {
 		inherit: true,
 		basePower: 1,
+		damageCallback(pokemon) {
+			const psywaveDamage = (this.random(0, (1.5 * pokemon.level)));
```
Comment (Zarel) [inline data/mods/gen1/moves.ts:605]:
```suggestion
			const psywaveDamage = (this.random(0, this.trunc(1.5 * pokemon.level)));
```

Comment (Zarel) [reply data/mods/gen1/moves.ts:605]: (`1.5 * pokemon.level` is rounded down before rolling the dice, right?)

File: data/mods/gen1/moves.ts [hunk 2]
```diff
@@ -601,6 +601,15 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+			if (psywaveDamage <= 0) {
+				this.hint("Desync Clause activated.");
+				return false;
+			} else {
+				return psywaveDamage;
+			}
```
Comment (Zarel) [inline data/mods/gen1/moves.ts:611]:
```suggestion
			if (psywaveDamage <= 0) {
				this.hint("Desync Clause activated.");
				return false;
			}
			return psywaveDamage;
```

File: data/mods/stadium/moves.ts [hunk 3]
```diff
@@ -55,6 +55,13 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 			},
 		},
 	},
+	psywave: {
+		inherit: true,
+		basePower: 1,
+		damageCallback(pokemon) {
+			return (this.random(1, (1.5 * pokemon.level)));
```
Comment (Zarel) [inline data/mods/stadium/moves.ts:62]:
```suggestion
			return this.random(1, this.trunc(1.5 * pokemon.level));
```

**Comment by Zarel** [comment]
I'll leave this open for @DaWoblefet or @Marty-D to comment, but I personally think this is good.

**Review by Zarel — APPROVED** [review]

**Comment by DaWoblefet** [comment]
No need for my approval with super oldgen mechanics, I'm no expert with those (like anything before Gen 5). You could write a test if you weren't sure, but if it looks good to you two that's fine with me.

**Comment by Zarel** [comment]
Tests are annoying with RNG mechanics, and these usually don't need tests anyway (the code describes what it's supposed to do well enough).

**Comment by Zarel** [comment]
We'll accept this as-is. Thanks!

**Pull request merged**